### ISSUE TITLE
Shorthand queries and mutations where result returns single field.

### ISSUE THREAD

**Issue by rafalp** [mirumee/ariadne-codegen#66]
Taking a query like this:

```graphql
query GetAuthenticatedUser {
  me {
    id
    username
  }
}
```

Codegen would generate a client method that would return `GetAuthenticatedUserUser` directly from the `me` field, so there's no need for:

```python
result = client.get_authenticated_user()
auth_user = result.me
```

Instead we would get:

```python
auth_user = client.get_authenticated_user()
```

We could have an setting like `unwrap_single_fields=false` for enabling this behavior.

**Comment by EDjur** [comment]
Hi! Thanks for taking time to think about this in such a short notice.

When exploring this initially I also gleaned over `generate_result_class` if I could somehow write a plugin that could help me to do what I want. I think your second approach makes a lot of sense and is non-invasive.

Do you see that being an "official" plugin that would sit in this repo or up to users to write themselves? It seems a relatively common use case so I guess a more official plugin makes sense?

**Comment by mat-sop** [comment]
In my opinion we can treat this plugin as "official" and include it in repo. Maybe using `unwrap_single_fiels` we can "hide" it (but still document how this flag works), in [settings](https://github.com/mirumee/ariadne-codegen/blob/main/ariadne_codegen/settings.py#L59) for example:
```py
if self.unwrap_single_fiels:
    self.plugins.insert(0, "ariadne_codegen.ShorterResultsPlugin")
```

@rafalp What do you think about including this plugin in repo and still using `unwrap_single_fiels` flag?

**Comment by rafalp** [comment]
There are features we want to do out of the box. But if those features are contributed to us as a plugin, I feel that we should treat them as a plugins and don't shortcut them under dedicated settings. Like how Ariadne and Django does some extra features as extensions and keeps them under `.contrib` package.

Doing it like this we get extra bonus of having real plugins in package for plugin devs to look  to for example and inspirations.

**Comment by mat-sop** [comment]
Resolved by #141 @bombsimon Thank you for contributing :heart:
